planner/core/task.go

```diff
@@ -554,7 +554,18 @@ func (p *PhysicalBroadCastJoin) GetCost(lCnt, rCnt float64) float64 {
 		rightSchema:   p.children[1].Schema(),
 	}
 	numPairs := helper.estimate()
+	if p.JoinType == SemiJoin || p.JoinType == AntiSemiJoin || p.JoinType == LeftOuterSemiJoin || p.JoinType == AntiLeftOuterSemiJoin {
+		if len(p.OtherConditions) > 0 {
+			numPairs *= 0.5
+		} else {
+			numPairs = 0
+		}
+	}
```

Comment:
like the left conditions and right conditions, other conditions only add some cost, but not to reduce probe cost. here refer to the tidb part.

Reply:
I don't get your point, the cost calculation for `PhysicalBroadCastJoin` is the same as `PhysicalHashJoin`.